specification/glossary.md

```diff
@@ -2,10 +2,130 @@
 
 *This section is non-normative.*
 
-`TODO` Needs to be completed and linked
+## Adjustment
 
-Resource:
-> Add definition
+A charge representing a modification to billing data to account for certain events or circumstances not previously captured, or captured incorrectly. Examples include: billing errors, service disruptions, or pricing changes.
 
-Service:
-> Add definition
+## Amortization
+
+The gradual recognition of upfront costs over the duration of time that the good or service is consumed. An upfront payment made toward a commitment based discount can be amortized over commitment life. Amortization is valuable when the commitment period (time duration of the cost) extends beyond the granularity of the source report.
+
+## Billed Cost
+
+The total amount of fees and discounts as documented in an itemized bill.
+
+## Billing Account
+
+A mandatory account established with a service provider to manage and pay for all charges associated with an invoice. A billing account might have sub-accounts, all of whose costs are consolidated to the billing account.
+
+## Billing Currency
+
+The specific currency in which invoices or bills for services are issued and payments are expected to be made.
+
+## Billing Period
+
+The time interval over which a customer's usage of services is measured for billing purposes. For most cloud providers, the billing period is a month.
+
+## Block Pricing
+
+Pricing model where the cost of a particular resource or service is determined based on predefined blocks or tiers of usage. In these scenarios, the Pricing Quantity can be different from the Usage Quantity.
```

Comment:
Let's not use "pricing model" since block pricing is a bit different than how we discuss pricing models.
Block pricing is associated with SKUs, not resources or services.
nit: Replacing "blocks" with "chunks" to use a different word. If you prefer blocks, that's also fine.
```suggestion
Pricing approach where the cost of a particular SKU is determined based on predefined chunks of units. When a SKU uses block pricing, the Pricing Quantity can be different from the Usage Quantity.
```